pennylane_lightning/src/util/Memory.hpp

```diff
@@ -80,7 +80,16 @@ template <class T> class AlignedAllocator {
     const uint32_t alignment_;
 
   public:
+    using pointer = T *;
+    using const_pointer = const T *;
```

Comment:
It is just removed from `std::allocator` but optional to define it for a custom allocator. see https://en.cppreference.com/w/cpp/named_req/Allocator#Allocator_completeness_requirements

Reply:
Yep, as @chaeyeunpark fine to keep it, but it is not default included. We will likely move this to the new standard allocator definitions at a future time, but I think this meets most compilers requirements right now.

Reply:
OK, I'm aware that it is just removed from `std::allocator`.
I'm wondering why we need to define it for our custom allocator if this is optional. 
Are the compilers looking for these?
Maybe because this allocator is not following the new standard yet, and because of that these are needed?

Reply:
@chaeyeunpark Can you try to remove all of the new additions to the reduced C++20 set, and see if the const/non const change fixed the original issue?

Reply:
Thank you for clarifying that.